### ISSUE TITLE
Separate legacy Mali driver but not the compiled dtb 

### ISSUE THREAD

**Issue by nijhawank** [hexdump0815/linux-mainline-and-mali-generic-stable-kernel#3]
I see that you now compile both legacy Mali and Lima driver, also package them separately but say that in order to use the legacy Mali driver we would need to use the dts patch (and I guess recompile). Is the understanding correct? 
What would be the use of pre-compiled Mali driver without the required dtb, if the user has to anyway go through the compilation process? Couldn’t you have compiled and packaged the dtb as required by Mali?

**Comment by hexdump0815** [comment]
i planned to actually make it work with only one dtb, so that it can easily switched in there, but it looks like the legacy mali has a few extra requirements and after a while of debugging i decided that it is not worth the effort and tried to find a somewhat clean approach to easily switch between mali and lima - its just one single and small patch now.

if you want lima the just use the normal tar.gz file (it includes lima) and if you want legacy mali, just use the legacy.tar.gz and the mali.tar.gz ... the packaging is done this way to make it easily useable for my image building frameworks without having to adjust too much.

i tested both - mali and lima - and both work very well ... so enjoy v5.10, which is an long term support kernel again :) ... btw. i plan to build some fresh images based on this kernel and some other nice ideas i have in mind in a few weeks.

**Comment by nijhawank** [comment]
So you are saying that the legacy.tar.gz already includes the dtb as required by legacy Mali driver? 
Apologies, your comments in the release notes gave the impression that while we have the compiled legacy Mali driver, we still need to manually patch and compile the dtb which was my question “why”. Let me know if the understanding is correct.

thanks for your work in keeping alive this old toy

**Comment by hexdump0815** [comment]
yes - if you want legacy mali, just use the legacy tar.gz combined with the mali .tar.gz - that should work ... if you want lima, just use the normal tar.gz ... sorry, the documentation is not yet perfect as usual :)
